### DISCUSSION TITLE
Is Eventlet Still The Best Option?

### DISCUSSION THREAD

**Discussion by ebram96** [miguelgrinberg/Flask-SocketIO#2068] [Q&A]
I'm quoting from documentation:
"[eventlet is the best performant option, with support for long-polling and WebSocket transports.](https://flask-socketio.readthedocs.io/en/latest/intro.html)"

But when I check Eventlet's documetation I see:
"[New usages of eventlet are now heavily discouraged!](https://eventlet.readthedocs.io/en/latest/)"

I'm wondering if I'm starting to use Flask-SocketIO in a new project, should I still go with Eventlet?
If no, what is the recommendation here?

**Comment by miguelgrinberg** [comment]
No, eventlet is not a good option anymore, it is a project that is winding down and will eventually be closed. For a new project I would recommend you start with the threading mode and if that is insufficient you switch to gevent.

**Reply by ebram96** [reply]
Thank you for your response. In suce case, it is necessary to update Flask-SocketIO's documentation to reflect the current best recommendations..
Also would you please let me know about the threading mode? I didn't find any mention of that in the documentation.

**Reply by miguelgrinberg** [reply]
Threading mode is achieved using `async_mode='threading`, or just by default if you don't have eventlet or gevent installed. It is mentioned in the documentation, and is even supported with Gunicorn using the `--threads` option. See the deployment section for details.
